# Jupyter notebook_David_Carvalho_Gabriel_Santos_Jonatan_Santana_Juliana_Valentim
O que está achando da atividade até aqui? *Estamos achando a atividade interessante! É uma boa oportunidade para aprendermos sobre análise de dados com Python e pandas.*


In [2]:
import pandas as pd

In [4]:
'as'

'as'

In [5]:
pd.read_excel

<function pandas.io.excel._base.read_excel(io, sheet_name: 'str | int | list[IntStrT] | None' = 0, *, header: 'int | Sequence[int] | None' = 0, names: 'SequenceNotStr[Hashable] | range | None' = None, index_col: 'int | str | Sequence[int] | None' = None, usecols: 'int | str | Sequence[int] | Sequence[str] | Callable[[str], bool] | None' = None, dtype: 'DtypeArg | None' = None, engine: "Literal['xlrd', 'openpyxl', 'odf', 'pyxlsb', 'calamine'] | None" = None, converters: 'dict[str, Callable] | dict[int, Callable] | None' = None, true_values: 'Iterable[Hashable] | None' = None, false_values: 'Iterable[Hashable] | None' = None, skiprows: 'Sequence[int] | int | Callable[[int], object] | None' = None, nrows: 'int | None' = None, na_values=None, keep_default_na: 'bool' = True, na_filter: 'bool' = True, verbose: 'bool' = False, parse_dates: 'list | dict | bool' = False, date_parser: 'Callable | lib.NoDefault' = <no_default>, date_format: 'dict[Hashable, str] | str | None' = None, thousands: 's

In [6]:
pip install pandas openpyxl

Note: you may need to restart the kernel to use updated packages.


In [29]:
df = pd.read_excel(r"C:\pasta1.xls")

In [30]:
pd.read_excel

<function pandas.io.excel._base.read_excel(io, sheet_name: 'str | int | list[IntStrT] | None' = 0, *, header: 'int | Sequence[int] | None' = 0, names: 'SequenceNotStr[Hashable] | range | None' = None, index_col: 'int | str | Sequence[int] | None' = None, usecols: 'int | str | Sequence[int] | Sequence[str] | Callable[[str], bool] | None' = None, dtype: 'DtypeArg | None' = None, engine: "Literal['xlrd', 'openpyxl', 'odf', 'pyxlsb', 'calamine'] | None" = None, converters: 'dict[str, Callable] | dict[int, Callable] | None' = None, true_values: 'Iterable[Hashable] | None' = None, false_values: 'Iterable[Hashable] | None' = None, skiprows: 'Sequence[int] | int | Callable[[int], object] | None' = None, nrows: 'int | None' = None, na_values=None, keep_default_na: 'bool' = True, na_filter: 'bool' = True, verbose: 'bool' = False, parse_dates: 'list | dict | bool' = False, date_parser: 'Callable | lib.NoDefault' = <no_default>, date_format: 'dict[Hashable, str] | str | None' = None, thousands: 's

In [31]:
display(df)

ID  LIMIT_BAL  SEX  EDUCATION  MARRIAGE  AGE PAY_1  PAY_2  \
0      798fc410-45c1      20000    2          2         1   24     2      2   
1      8a8c8f3b-8eb4     120000    2          2         2   26    -1      2   
2      85698822-43f5      90000    2          2         2   34     0      0   
3      0737c11b-be42      50000    2          2         1   37     0      0   
4      3b7f77cc-dbc0      50000    1          2         1   57    -1      0   
...              ...        ...  ...        ...       ...  ...   ...    ...   
29995  ecff42d0-bdc6     220000    1          3         1   39     0      0   
29996  99d1fa0e-222b     150000    1          3         2   43    -1     -1   
29997  95cdd3e7-4f24      30000    1          2         2   37     4      3   
29998  00d03f02-04cd      80000    1          3         1   41     1     -1   
29999  15d69f9f-5ad3      50000    1          2         1   46     0      0   

       PAY_3  PAY_4  ...  BILL_AMT4  BILL_AMT5  BILL_AMT6  PAY_AMT1  PAY_AMT2  \
0         -1     -1  ...          0          0          0         0       689   
1          0      0  ...       3272       3455       3261         0      1000   
2          0      0  ...      14331      14948      15549      1518      1500   
3          0      0  ...      28314      28959      29547      2000      2019   
4         -1      0  ...      20940      19146      19131      2000     36681   
...      ...    ...  ...        ...        ...        ...       ...       ...   
29995      0      0  ...      88004      31237      15980      8500     20000   
29996     -1     -1  ...       8979       5190          0      1837      3526   
29997      2     -1  ...      20878      20582      19357         0         0   
29998      0      0  ...      52774      11855      48944     85900      3409   
29999      0      0  ...      36535      32428      15313      2078      1800   

       PAY_AMT3  PAY_AMT4  PAY_AMT5  PAY_AMT6  default payment next month  
0             0         0         0         0                           1  
1          1000      1000         0      2000                           1  
2          1000      1000      1000      5000                           0  
3          1200      1100      1069      1000                           0  
4         10000      9000       689       679                           0  
...         ...       ...       ...       ...                         ...  
29995      5003      3047      5000      1000                           0  
29996      8998       129         0         0                           0  
29997     22000      4200      2000      3100                           1  
29998      1178      1926     52964      1804                           1  
29999      1430      1000      1000      1000                           1  

[30000 rows x 25 columns]

In [32]:
df.shape


(30000, 25)

1- Saber quantas colunas os dados contêm?
Neste arquivo de dados existem 25 colunas.

2- Quantas linhas (amostras)?
Neste arquivo, há 30000 linhas. Assim, 30000 amostras foram recolhidas.

3- Que tipos de características existem. Quais são categóricas e quais são numéricas? (Os valores das características categóricas pertencem a classes discretas como “sim”, “não” ou “talvez”. Normalmente as características numéricas pertencem a uma escala numérica contínua, como as quantias em dólares).

In [33]:
df_categorica = df[['SEX', 'EDUCATION', 'MARRIAGE', 'PAY_1', 'PAY_2', 'PAY_3', 'PAY_4', 'PAY_5', 'PAY_6', 'default payment next month']]
display(df_categorica)

SEX  EDUCATION  MARRIAGE PAY_1  PAY_2  PAY_3  PAY_4  PAY_5  PAY_6  \
0        2          2         1     2      2     -1     -1     -2     -2   
1        2          2         2    -1      2      0      0      0      2   
2        2          2         2     0      0      0      0      0      0   
3        2          2         1     0      0      0      0      0      0   
4        1          2         1    -1      0     -1      0      0      0   
...    ...        ...       ...   ...    ...    ...    ...    ...    ...   
29995    1          3         1     0      0      0      0      0      0   
29996    1          3         2    -1     -1     -1     -1      0      0   
29997    1          2         2     4      3      2     -1      0      0   
29998    1          3         1     1     -1      0      0      0     -1   
29999    1          2         1     0      0      0      0      0      0   

       default payment next month  
0                               1  
1                               1  
2                               0  
3                               0  
4                               0  
...                           ...  
29995                           0  
29996                           0  
29997                           1  
29998                           1  
29999                           1  

[30000 rows x 10 columns]

In [34]:
df_numerica = df[['LIMIT_BAL','AGE', 'BILL_AMT1', 'BILL_AMT2', 'BILL_AMT3', 'BILL_AMT4', 'BILL_AMT5', 'BILL_AMT6']]
display(df_numerica)

LIMIT_BAL  AGE  BILL_AMT1  BILL_AMT2  BILL_AMT3  BILL_AMT4  BILL_AMT5  \
0          20000   24       3913       3102        689          0          0   
1         120000   26       2682       1725       2682       3272       3455   
2          90000   34      29239      14027      13559      14331      14948   
3          50000   37      46990      48233      49291      28314      28959   
4          50000   57       8617       5670      35835      20940      19146   
...          ...  ...        ...        ...        ...        ...        ...   
29995     220000   39     188948     192815     208365      88004      31237   
29996     150000   43       1683       1828       3502       8979       5190   
29997      30000   37       3565       3356       2758      20878      20582   
29998      80000   41      -1645      78379      76304      52774      11855   
29999      50000   46      47929      48905      49764      36535      32428   

       BILL_AMT6  
0              0  
1           3261  
2          15549  
3          29547  
4          19131  
...          ...  
29995      15980  
29996          0  
29997      19357  
29998      48944  
29999      15313  

[30000 rows x 8 columns]

Características Categóricas: São 10 colunas categóricas, sendo elas: SEX EDUCATION MARRIAGE PAY_1 PAY_2 PAY_3 PAY_4 PAY_5 PAY_6 E default payment next month

Características Numéricas: São 8 colunas numéricas, sendo elas: LIMIT_BAL AGE BILL_AMT1 BILL_AMT2 BILL_AMT3 BILL_AMT4 BILL_AMT5 E BILL_AMT6

4- Qual é a aparência dos dados segundo essas características. (Para saber isso, você pode examinar o intervalo de valores em características numéricas ou a frequência de classes diferentes em características categóricas, por exemplo).

In [35]:
df_categorica.describe()

SEX     EDUCATION      MARRIAGE         PAY_2         PAY_3  \
count  30000.000000  30000.000000  30000.000000  30000.000000  30000.000000   
mean       1.586500      1.833333      1.535900     -0.132867     -0.164333   
std        0.513348      0.807699      0.542698      1.191215      1.191096   
min        0.000000      0.000000      0.000000     -2.000000     -2.000000   
25%        1.000000      1.000000      1.000000     -1.000000     -1.000000   
50%        2.000000      2.000000      2.000000      0.000000      0.000000   
75%        2.000000      2.000000      2.000000      0.000000      0.000000   
max        2.000000      6.000000      3.000000      8.000000      8.000000   

              PAY_4         PAY_5         PAY_6  default payment next month  
count  30000.000000  30000.000000  30000.000000                30000.000000  
mean      -0.219300     -0.263900     -0.288033                    0.218733  
std        1.162348      1.127519      1.144981                    0.413394  
min       -2.000000     -2.000000     -2.000000                    0.000000  
25%       -1.000000     -1.000000     -1.000000                    0.000000  
50%        0.000000      0.000000      0.000000                    0.000000  
75%        0.000000      0.000000      0.000000                    0.000000  
max        8.000000      8.000000      8.000000                    1.000000

5)Existem dados faltando?

In [36]:
(df == "Not available").sum()


ID                               0
LIMIT_BAL                        0
SEX                              0
EDUCATION                        0
MARRIAGE                         0
AGE                              0
PAY_1                         3021
PAY_2                            0
PAY_3                            0
PAY_4                            0
PAY_5                            0
PAY_6                            0
BILL_AMT1                        0
BILL_AMT2                        0
BILL_AMT3                        0
BILL_AMT4                        0
BILL_AMT5                        0
BILL_AMT6                        0
PAY_AMT1                         0
PAY_AMT2                         0
PAY_AMT3                         0
PAY_AMT4                         0
PAY_AMT5                         0
PAY_AMT6                         0
default payment next month       0
dtype: int64

5- R: Ao acessar o banco de dados, nota-se a presença da palavra "Not Available", que nesta planilha se refere aos valores NULOS. Ao contar a quantidade de vezes que essa palavra aparece na planilha, percebe-se que ela aparece 3021 vezes na coluna "PAY_1". Portanto, em 3021 linhas das 30000 amostras, existem informações ausentes relacionadas ao primeiro pagamento.